src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs

```diff
@@ -2744,11 +2745,9 @@ public static void ObjectEnumeratorIndependentWalk()
                     Assert.Equal(test, property.Value.GetInt32());
                     test++;
 
-                    // Subsequent read of the same JsonProperty doesn't allocate a new string
-                    // (if another property is inspected from the same document that guarantee
-                    // doesn't hold).
+                    // Subsequent read of the same JsonProperty should return an equal string
                     string propertyName2 = property.Name;
-                    Assert.Same(propertyName, propertyName2);
```

Comment:
Do we know why this was optimized in the first place?

Reply:
See https://github.com/dotnet/runtime/pull/76450#issuecomment-1264839881